src/prefs.ts

```diff
@@ -77,6 +77,12 @@ class RandomWallpaperSettings extends ExtensionPreferences {
             comboZoomMode,
             'selected',
             Gio.SettingsBindFlags.DEFAULT);
+        settings.observe('zoom-mode', () => {
+            new Settings.Settings('org.gnome.desktop.background')
+                .setString('picture-options', WallpaperManager.getZoomModeEnum()[settings.getInt('zoom-mode')]);
+            new Settings.Settings('org.gnome.desktop.screensaver')
+                .setString('picture-options', WallpaperManager.getZoomModeEnum()[settings.getInt('zoom-mode')]);
+        });
```

Comment:
With the above things in mind, this also has the same problem.
This forces the `spanned` mode for multi-monitor setups back into a single screen wallpaper.